### PR TITLE
Fix optimize_for_mobile to prevent it from producing a module with a missing bias when the module contains a convolution without bias followed by batch normalization.

### LINKED ISSUE #76431: Torch.utils.mobile_optimizer.optimize_for_mobile is resulting different output than torch model and jit model
### 🐛 Describe the bug

When I convert torch model for mobile optimizer, it is giving different outputs. But the output is the same for torch model and jit model.

```
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

# model definition with two classes
model = torchvision.models.mobilenet_v3_large()
model.classifier[3] = torch.nn.Linear(in_features=1280, out_features=2)
model.load_state_dict(torch.load('model.ptl'))
model.eval()
```

```
# Optimize model for mobile

example = torch.rand(1, 3, 224, 224)

traced_script_module = torch.jit.trace(model, example)

traced_script_module_optimized = optimize_for_mobile(traced_script_module)

```
1. INFERENCE TORCH MODEL
```
model.forward(torch.ones([1, 3, 224, 224], dtype=torch.float))

# Output
tensor([[-3.0561,  3.0894]], grad_fn=<AddmmBackward0>)
```
2. INFERENCE JIT MODEL
```
traced_script_module.forward(torch.ones([1, 3, 224, 224], dtype=torch.float))

# Output
tensor([[-3.0561,  3.0894]], grad_fn=<AddmmBackward0>)
```
3. INFERENCE OPTIMZED MODEL
```
traced_script_module_optimized.forward(torch.ones([1, 3, 224, 224], dtype=torch.float))

# Output
tensor([[-4.5466,  6.5033]])
```
I tested with several examples. The same result: after optimizing for mobile, the model's output is different

### Versions

```
Collecting environment information...
                                                                                  
PyTorch version: 1.11.0                                                                                                 
CUDA used to build PyTorch: 11.3                                                                                                                                                                                                                                                                                                      
OS: Microsoft Windows 10 Pro                                                                                            
                                                                                  
                                                                                                                                                                                                                           
Python version: 3.9.7                                                                        
Is CUDA available: True                                                                                                 
CUDA runtime version: 11.2.152                                                                                                                                                                                                                                                                                                               
Nvidia driver version: 461.33                                                                                           
```

### LINKED ISSUE #76807: There is a bug with latest stable torch version and the following Nightly versions related to `optimize_for_mobile`
### 🐛 Describe the bug

There is a bug in `optimize_for_mobile` with latest torch version and the following nightly versions. Please look at this [discussion](https://discuss.pytorch.org/t/torch-utils-mobile-optimizer-optimize-for-mobile-is-resulting-different-output-than-torch-model-and-jit-model/150183) for full details.

**The short problem description:** the model is giving different result after `optimize_for_mobile`

The problem started  with stable torch version 1.11.0 and the following Nightly versions. But it is working correctly with previous torch versions. 

Here is the minimal code snippet to reproduce the error:

```
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

model = torchvision.models.resnet50(pretrained=True)
model.eval()


out1 = model(torch.ones([1, 3, 224, 224], dtype=torch.float))
out2 = traced_script_module(torch.ones([1, 3, 224, 224], dtype=torch.float))
out3 = traced_script_module_optimized(torch.ones([1, 3, 224, 224], dtype=torch.float))
```
and results:

```
tensor([[-3.0806e-01,  7.9845e-02, -1.1900e+00, -1.4837e+00, -5.1359e-01,...]])
tensor([[-3.0806e-01,  7.9845e-02, -1.1900e+00, -1.4837e+00, -5.1359e-01,...]])
tensor([[-5.2194e+00,  1.4939e+00, -2.2703e+00, -5.4212e+00, -2.0551e+00,...]])
```

### Versions

 torch version 1.11.0 and the following Nightly versions.

### LINKED ISSUE #116515: mobile model has very different inference result
### 🐛 Describe the bug

Step1. Generate the mobile optimized model:
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

model = torchvision.models.mobilenet_v3_small(pretrained=True)
model.eval()
example = torch.rand(1, 3, 224, 224)
traced_script_module = torch.jit.trace(model, example)
optimized_traced_model = optimize_for_mobile(traced_script_module)
optimized_traced_model._save_for_lite_interpreter("mobilenet_v3_small.ptl")

Step2. Use this model to run an inference, using a husky raw image (attached as zip file)
import torch
import numpy as np
from torch.jit import mobile
mobilenet = mobile._load_for_lite_interpreter("model.mobilenet_v3_small.ptl")

input = np.fromfile("imagenet_husky.raw", dtype=np.float32)
input = input.reshape((1, 3,224,224))
intensor = torch.from_numpy(input)
out = mobilenet(intensor)
_, index = torch.max(out, 1) 

You get index = 574, which does not make sense.

Step3. Use the original mobilenet model to run the same input, and get correct result

from torchvision import models
import torch
import numpy as np
mobilenet = models.mobilenet_v3_small(pretrained=True)
mobilenet.eval()

input = np.fromfile("imagenet_husky.raw", dtype=np.float32)
input = input.reshape((1, 3,224,224))
intensor = torch.from_numpy(input)
out = mobilenet(intensor)
_, index = torch.max(out, 1)

You get index = 250, which is 'Eskimo dog, husky', correct result.

[husky01.zip](https://github.com/pytorch/pytorch/files/13788963/husky01.zip)

### Versions

Collecting environment information...
PyTorch version: 2.0.1+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun  4 2021, 15:09:15)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-88-generic-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 Ti
GPU 1: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 470.223.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 48 bits virtual
CPU(s):                             12
On-line CPU(s) list:                0-11
Thread(s) per core:                 2
Core(s) per socket:                 6
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              79
Model name:                         Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz
Stepping:                           1
CPU MHz:                            2863.668
CPU max MHz:                        4000.0000
CPU min MHz:                        1200.0000
BogoMIPS:                           7195.26
L1d cache:                          192 KiB
L1i cache:                          192 KiB
L2 cache:                           1.5 MiB
L3 cache:                           15 MiB
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        KVM: Mitigation: VMX unsupported
Vulnerability L1tf:                 Mitigation; PTE Inversion
Vulnerability Mds:                  Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:             Mitigation; PTI
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; Clear CPU buffers; SMT vulnerable
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts md_clear flush_l1d

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==2.0.1+cpu
[pip3] torchaudio==2.0.2+cpu
[pip3] torchvision==0.15.2+cpu
[conda] numpy                     1.24.1                   pypi_0    pypi
[conda] torch                     2.0.1+cpu                pypi_0    pypi
[conda] torchaudio                2.0.2+cpu                pypi_0    pypi
[conda] torchvision               0.15.2+cpu               pypi_0    pypi

### PR DESCRIPTION
**Pull request by AkiSakurai**
OptimizeForMobile might convert _convolution to conv2d, making the assumption of replaceConvBiasWithGetAttr invalid. https://github.com/pytorch/pytorch/blob/a32eac345f5d7862246c397d0a32b5e0cbca77ff/torch/csrc/jit/passes/xnnpack_rewrite.cpp#L427 https://github.com/pytorch/pytorch/blob/a32eac345f5d7862246c397d0a32b5e0cbca77ff/torch/csrc/jit/passes/fold_conv_bn.cpp#L50-L52

fix #76807, fix #116515, fix #76431

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
LGTM, but before landing do you think you could add a bit more info to the summary about what breaks prior to this PR? Does the conv-batchnorm folding later on assume that the bias has been replaced with a getattr, and does it just produce wrong results instead of erroring out when this occurs?

(not necessarily asking you to fix this, mostly just curious what the current state is)
